library/core/src/ptr/non_null.rs

```diff
@@ -1752,6 +1752,9 @@ impl<T: ?Sized> From<&mut T> for NonNull<T> {
     /// Converts a `&mut T` to a `NonNull<T>`.
     ///
     /// This conversion is safe and infallible since references cannot be null.
+    ///
+    /// It is semantically equivalent to `NonNull::new_unchecked(reference as *mut T)`, including
+    /// with respect to provenance.
```

Comment:
This would be the first mention of provenance in the stable part of the docs. So to land it as-is, it should be blocked on https://github.com/rust-lang/rust/pull/130350 (and should add a link to the new "provenance" section which that PR adds).

I guess this is nit super urgent so it may be fine to be blocked? Alternatively, you can remove the mention of provenance, and I can rebase https://github.com/rust-lang/rust/pull/130350 once your PR lands, and add it back there.

Reply:
Agreed, totally fine for this to be blocked on https://github.com/rust-lang/rust/pull/130350. I'll follow up once it's landed.

Reply:
https://github.com/rust-lang/rust/pull/130350 has landed. :) @joshlf can you reword accordingly?